### ISSUE TITLE
write_database replace table not working

### ISSUE THREAD

**Issue by aklaver** [pola-rs/polars#12684]
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```
import polars as pl
uri = "postgres://postgres:postgres@localhost:5462/test"
df = pl.read_parquet('/home/aklaver/Downloads/yellow_tripdata_2023-09.parquet')
df.write_database('taxi_test', uri, engine='adbc', if_exists='replace')
df.write_database('taxi_test', uri, engine='adbc', if_exists='replace')
ProgrammingError: INVALID_ARGUMENT: [libpq] Failed to create table: ERROR:  relation "taxi_test" already exists
```
In Postgres log:
```
CREATE TABLE "public" . "taxi_test"
ERROR:  relation "taxi_test" already exists``
```
There is no DROP TABLE "public" . "taxi_test" before the CREATE TABLE.

Assuming this is supposed to follow the Pandas behavior it should DROP the table first, otherwise I don't see the point.

### Issue description

write_database(if_exists='replace') should DROP the table before doing the CREATE TABLE.

### Expected behavior

See issue description.

### Installed versions

<details>

```
--------Version info---------
Polars:              0.19.16
Index type:          UInt32
Platform:            Linux-6.2.0-37-generic-x86_64-with-glibc2.35
Python:              3.9.18 (main, Aug 25 2023, 13:20:14) 
[GCC 11.4.0]

----Optional dependencies----
adbc_driver_sqlite:  <not installed>
cloudpickle:         <not installed>
connectorx:          0.3.2
deltalake:           <not installed>
fsspec:              <not installed>
gevent:              <not installed>
matplotlib:          <not installed>
numpy:               1.23.5
openpyxl:            3.0.10
pandas:              2.0.1
pyarrow:             11.0.0
pydantic:            <not installed>
pyiceberg:           <not installed>
pyxlsb:              <not installed>
sqlalchemy:          2.0.13
xlsx2csv:            <not installed>
xlsxwriter:          <not installed>

```
Also:
```
adbc-driver-manager==0.8.0
adbc-driver-postgresql==0.8.0
```
</details>

**Comment by aklaver** [comment]
Realized pl.show_versions() did not include:
```
adbc-driver-manager==0.8.0
adbc-driver-postgresql==0.8.0
```

**Comment by alexander-beedie** [comment]
Was able to replicate locally; looks like we are setting the `mode` flag incorrectly here.
I'll get it fixed later this afternoon 👌

**Comment by aklaver** [comment]
Thank you.

**Comment by alexander-beedie** [comment]
@aklaver: The update is now available in the new release, `0.19.18` 👌

**Comment by aklaver** [comment]
Great, I will update. Thanks.
